integrals: Improved heurisch.py to get Unevaluated Integral Object of som…

wrat opened the pull request:
Integral: Updated heurisch.py to get Unevaluated Integral Object of Complex Expression

As we Know From Documentation That If ingrate() funtion is unable to solve any Integrand then it return its Unevaluated Integral Object .
But for Some Expression like x, log(x), xlog(x), log(xlog(x)), xlog(xlog(x)) ,xlog(xlog(x**log(x))),etc. so While Caculating Integration of this Expression heurisch.py raise a ValueError Execption from " numer = ring.from_expr(raw_numer) " .
As we can see these expression are not polynomial so code have to raise a PolynomialError which is caught by _eval_integral .
So I have changed some in code by which a PolynomialError raise and _eval_integral return a Unevaluated integral Object.

```

>>> from sympy import*
>>> from sympy import init_printing
>>> init_printing(use_unicode=True)
>>> x=symbols('x')
>>> expr=x**log(x**log(x**log(x)))
>>> expr#shows proper output
>>> integrate(expr,x)#shows proper output not an exception

```

 so as we can see Now we able to get unevaluated Integral Object.
 Previously it showing Error, issuse #10680.

wrat committed:
integrals: Improved heurisch.py to Unevalauted Integral Object of some Complex Epression
As we Know From Documentation That If ingrate() funtion is unable to solve any Integrand then it return its Unevaluated Integral Object .
But for Some Expression like x, log(x), xlog(x), log(xlog(x)), xlog(xlog(x)) ,xlog(xlog(x**log(x))),etc. so While Caculating Integration of this Expression heurisch.py raise a ValueError Execption from " numer = ring.from_expr(raw_numer) " .
As we can see these expression are not polynomial so code have to raise a PolynomialError which is caught by _eval_integral .
So I have changed some in code by which a PolynomialError raise and _eval_integral return a Unevaluated integral Object.
>>> from sympy import*
>>> from sympy import init_printing
>>> init_printing(use_unicode=True)
>>> x=symbols('x')
>>> expr=x**log(x**log(x**log(x)))
>>> expr#shows proper output
>>> integrate(expr,x)#shows proper output not an exception
>>>
 so as we can see Now we able to get unevaluated Integral Object.
 Previously it showing Error, issuse #10680.

jksuom commented on sympy/integrals/tests/test_heurisch.py:
```diff
@@ -71,7 +71,9 @@ def test_heurisch_exp():
 
-
+def test_issue_10680():
+    assert integrate(x**log(x**log(x**log(x))),x)
```
This could be

``assert isinstance(integrate(x**log(x**log(x**log(x))),x), Integral)``

to show that no error occurs and an unevaluated integral is returned.

You can edit this and then commit and push it to github. It will be automatically added to the pull request. (Run tests again before pushing.)

wrat committed:
test cases is added for issue #10680
 " def test_issue_10680()
     assert isinstance(integrate(x**log(x**log(x**log(x))),x), Integral)

"

wrat committed:
Imported the definition of integrate funtion
" from sympy.integrals.integrals import integrate"

wrat committed:
Merge due to changes
Merge branch 'master' into 1876_branch

jksuom commented on sympy/integrals/tests/test_heurisch.py:
```diff
@@ -71,7 +73,9 @@ def test_heurisch_exp():
     assert heurisch(Sum(x**z, (z, 1, 2)).function, z) == x**z/log(x)
 
-
+def test_issue_10680():
```
This function was also defined above. It should not be repeated.

wrat committed:
Removed Redefinition of "
 def test_issue_10680():
"

wrat committed:
"Removed Trailing whitespace from line 7 in
sympy/integrals/tests/test_heurisch.py"

wrat committed:
"Removed Trailing whitspace from line 75"

jksuom commented:
It is not necessary to close and reopen after each change. It will suffice to just push new commits to github.

wrat commented:
@jksuom  ping!!

jksuom commented:
I think this is ready now. Thanks!
